src/server/pfs/server/transaction_access.go

```diff
@@ -0,0 +1,43 @@
+package server
+
+import (
+	"bytes"
+	"context"
+
+	"github.com/pachyderm/pachyderm/v2/src/pps"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/storage/fileset"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal"
+	"github.com/pachyderm/pachyderm/v2/src/internal/transactionenv/txncontext"
+	"github.com/pachyderm/pachyderm/v2/src/pfs"
+)
+
+type FileAccessor struct {
+	ctx       context.Context
+	apiServer pfs.APIServer
+	d         *driver
+}
+
+func (a *apiServer) NewFileAccessor(ctx context.Context) txncontext.FileAccessor {
+	return &FileAccessor{
+		ctx:       ctx,
+		apiServer: a,
+		d:         a.driver,
+	}
+}
+
+// CreateFileset creates a fileset with a single file
```

Comment:
If we are going to fix this in a later P.R. We should have a TODO here indicating that this is still a problem.

Reply:
We also don't need the PFS driver in here.  We can just use the APIServer.

Reply:
I was hoping to avoid writing another grpc streaming server wrapper

Reply:
You need to send one message over the CreateFileset stream, then close it.  The generated client should have the same API as ModifyFile.  You may be able to factor something out into clientsdk that client and this can reference.  But it shouldn't be much code either way to create a fileset with one file.

Reply:
The requests aren't the issue, but the client is only generated automatically when you actually call the gRPC, if I'm not mistaken. I had written the fake server logic in #6593 but you didn't seem to like that idea.

Reply:
In the case of that other PR: If you want a utility function that takes a slice of file specs or maybe a `map[string][]byte` and turns it into a fileset, that's fine.  I just think it should live somewhere other than the root of internal.

In the case of this P.R: It looks like we are just calling createFileset on the driver in a way we could also call it on the APIServer.  If there is a parameter or behavior that would be different between the two, then lets keep using the driver.

The common concern is that we put helper code in a discoverable place, only write it once, and only write it if the API doesn't provide an easy way to do it, and write it to use the gRPC API, so it will see the most use.